content/en/blog/_posts/2024-05-20-cloud-provider-migration.md

```diff
@@ -9,55 +9,57 @@ author: >
   Michael McCune (Red Hat)
 ---
 
-Since as early as Kubernetes v1.7, the Kubernetes project has pursued the ambitious goal of removing built-in cloud provider integrations.
+Since as early as Kubernetes v1.7, the Kubernetes project has pursued the ambitious goal of removing built-in cloud provider integrations ([KEP-2395](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md)).
 While these integrations were instrumental in Kubernetes' early development and growth, their removal was driven by two key factors:
 the growing complexity of maintaining native support for every cloud provider across millions of lines of Go code, and the desire to establish
 Kubernetes as a truly vendor-neutral platform.
 
 After many releases, we're thrilled to announce that all cloud provider integrations have been successfully migrated from the core Kubernetes repository to external plugins.
+In addition to achieving our initial objectives, we've also significantly streamlined Kubernetes by removing roughly 1.5 million lines of code and reducing the binary sizes of core components by approximately 40%.
```

Comment:
The 1.5 mililion lines removed is based on the lines removed from these 4 PRs:
https://github.com/kubernetes/kubernetes/pull/124519
https://github.com/kubernetes/kubernetes/pull/115838
https://github.com/kubernetes/kubernetes/pull/122937
https://github.com/kubernetes/kubernetes/pull/122857

And the 40% binary size reduction is from this comment: https://github.com/kubernetes/kubernetes/issues/76506#issuecomment-733404728 

I've also compared binary sizes from master and previous releases and saw similar comparisons.

Reply:
Wow